src/parser.cpp

```diff
@@ -356,22 +360,22 @@ namespace ada::parser {
       return false;
     }
 
-    out = ada::serializers::ipv6(address);
+    out = ada::serializers::ipv6(address.data());
     return true;
   }
 
   /**
    * @see https://url.spec.whatwg.org/#host-parsing
    */
-  bool parse_host(std::optional<std::string>& out, const std::string_view input, bool is_not_special, bool input_is_ascii) {
+  bool parse_host(std::optional<std::string>& out, const std::string_view input, bool is_not_special) {
     //
     // Note: this function assumes that parse_host is not empty. Make sure we can
     // guarantee that.
     //
     // If input starts with U+005B ([), then:
     if (input[0] == '[') {
       // If input does not end with U+005D (]), validation error, return failure.
-      if (input.back() != ']') {
+      if (input[input.size()-1] != ']') {
```

Comment:
This does not seem like an obvious gain. input.back() was clear enough.

Reply:
one microbenchmark showed that this change made a difference. happy to change it.